dsb-util/src/main/java/com/spectralogic/dsbrowser/util/FunctionalExtensions.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * ***************************************************************************
+ *   Copyright 2014-2017 Spectra Logic Corporation. All Rights Reserved.
+ *   Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ *   this file except in compliance with the License. A copy of the License is located at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file.
+ *   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ *   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *   specific language governing permissions and limitations under the License.
+ * ***************************************************************************
+ */
+
+package com.spectralogic.dsbrowser.util
+
+inline fun <T, R> ((T) -> R).andThen(crossinline block: () -> Unit): (T) -> R {
```

Comment:
You should add some javadoc to explain what this method does, specifically the inlining behavior of it as well.